lib/streamlit/runtime/fragment.py

```diff
@@ -119,43 +119,109 @@ def fragment(
     *,
     run_every: int | float | timedelta | str | None = None,
 ) -> Callable[[F], F] | F:
-    """Allow a function to be run independently of the full script.
-
-    Functions decorated with ``@st.experimental_fragment`` are handled specially within
-    an app: when a widget created within an invocation of the function (a fragment) is
-    interacted with, then only that fragment is rerun rather than the full streamlit app.
+    """Decorator to turn a function into a fragment which can rerun independently\
+    of the full script.
+
+    When a user interacts with an input widget created by a fragment, Streamlit
```

Comment:
I think I slightly prefer "created INSIDE a fragment" vs "created BY a fragment". See further comment on this below for more detailed discussion.

Reply:
I still prefer "created inside a fragment"